### PR TITLE
Fix correctness bug in runner V1.

### PR DESCRIPTION
**Pull request by py4**
# Description

There were two problems in the previous state of the code:
1. We calculate logits for all tokens (inefficient)
2. Because we calculate logits for all tokens, the logic for reading next token selects from index 0 and problematic  with (#1) 

In this PR, we select hidden states at the positions that we want to sample from. It makes the sampling cheaper and also fixes the correctness issue (#2)

Before:
```
--------------------------------------------------
Prompt: 'Hello, my name is'
Generated text: 'Question I am a 25-year-old woman from the United States. I am'
--------------------------------------------------
Prompt: 'The capital of France is'
Generated text: 'Question The city is known for its stunning architecture, art museums, fashion, and'
--------------------------------------------------
Prompt: 'The colors of the rainbow are'
Generated text: 'Question Red, Orange, Yellow, Green, Blue, Indigo, Violet.'
--------------------------------------------------
Prompt: 'The future of AI is'
Generated text: 'Question The future of AI is! The future of AI is!\nI, beauty'
--------------------------------------------------
Prompt: 'The president of the United States is'
Generated text: 'Question The president of the United States is! The president of Of the United States'
--------------------------------------------------
```
After:
```
--------------------------------------------------
Prompt: 'Hello, my name is'
Generated text: ' Emily and I am a 25-year-old freelance writer and editor. I have'
--------------------------------------------------
Prompt: 'The capital of France is'
Generated text: ' a city of romance, art, fashion, and cuisine. Paris is a must'
--------------------------------------------------
Prompt: 'The colors of the rainbow are'
Generated text: ' often remembered using the acronym ROYGBIV, which stands for Red, Orange'
--------------------------------------------------
Prompt: 'The future of AI is'
Generated text: ' bright, but it also raises concerns about bias, accountability, and the potential for'
--------------------------------------------------
Prompt: 'The president of the United States is'
Generated text: ' the head of state and head of government of the United States. The president serves'
--------------------------------------------------
```

# Tests

local

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Comment by fenghuizhang** [comment]
Thanks for fixing the code!

**Comment by fenghuizhang** [comment]
LGTM.

**Comment by xiangxu-google** [comment]
Thank you for fixing this!
